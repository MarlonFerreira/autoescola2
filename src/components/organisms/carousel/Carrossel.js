import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import "react-responsive-carousel/lib/styles/carousel.min.css"; // requires a loader
import { Carousel } from "react-responsive-carousel";

const Root = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  color: #fff;
  /* padding: 100px 0; */
  height: auto;
`;

const Carrossel = ({ number, imagens }) => {
  return (
    <>
      <Root>
        <Carousel
          infiniteLoop={true}
          showThumbs={false}
          centerMode
          centerSlidePercentage={number}
          autoPlay={true}
          transitionTime={500}
          interval={10000}
          statusFormatter={(current, total) => ``}
        >
          {imagens.map((imagem) => (
            <div key={imagem.id}>
              <img src={imagem.imagem} alt={imagem.legenda} />
              <p className="legend">{imagem.legenda}</p>
            </div>
          ))}
        </Carousel>
      </Root>
    </>
  );
};

Carrossel.defaultProps = {
  number: 100,
  imagens: [],
};

Carrossel.propTypes = {
  number: PropTypes.number,
  imagens: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      imagem: PropTypes.string,
      legenda: PropTypes.string,
      slang: PropTypes.string,
    })
  ),
};

export default Carrossel;
