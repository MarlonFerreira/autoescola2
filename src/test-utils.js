import React from "react";
import { render } from "@testing-library/react";
import ThemeProvider from "styles/themeProvider/ThemeProvider";
import { MemoryRouter as Router } from "react-router-dom";

const TesteProvider = ({ children }) => (
  <Router>
    <ThemeProvider>{children}</ThemeProvider>
  </Router>
);

export * from "@testing-library/react";

const customRender = (ui, options) =>
  render(ui, { wrapper: TesteProvider, ...options });

export { customRender as render };
