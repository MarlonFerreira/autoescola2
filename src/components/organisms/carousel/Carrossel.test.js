import React from "react";
import { render, screen } from "test-utils";

import Carrossel from "./Carrossel";

import image1 from "assets/serv-01.jpg";

test("Carrossel - match snapshot", () => {
  const { asFragment } = render(<Carrossel></Carrossel>);

  expect(asFragment()).toMatchSnapshot();
});

// (If this is intentional, then use the `*AllBy*` variant of the query (like `queryAllByText`, `getAllByText`, or `findAllByText`)).

test("Carrosel - Renderiza as imagens", async () => {
  const opcoes = [
    { id: 1, imagem: image1, legenda: "Imagem 1" },
    { id: 2, imagem: image1, legenda: "Imagem 2" },
    { id: 3, imagem: image1, legenda: "Imagem 3" },
    { id: 4, imagem: image1, legenda: "Imagem 4" },
    { id: 5, imagem: image1, legenda: "Imagem 5" },
    { id: 6, imagem: image1, legenda: "Imagem 6" },
  ];

  render(<Carrossel opcoes={opcoes}></Carrossel>);

  expect(await screen.getByText("Imagem 4")).toBeInTheDocument();
});
