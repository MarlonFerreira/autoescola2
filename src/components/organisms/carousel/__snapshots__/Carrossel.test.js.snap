// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carrossel - match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: #fff;
  padding: 100px 0;
  height: auto;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;
